src/comm.c

```diff
@@ -62,8 +63,13 @@ static void request_update() {
 static void in_received_handler(DictionaryIterator *received, void *context) {
   phone_contact = true;
   update_in_progress = false;
-  schedule_update(UPDATE_FREQUENCY);
-
+  Tuple* recency = dict_find(received, APP_KEY_RECENCY);
+  if (recency) {
```

Comment:
There are [3 types](https://github.com/mddub/nightscout-graph-pebble/blob/77c70a/src/app_keys.h#L7) of content for the app message, and they refer to the same indices in `received` in different ways. `APP_KEY_RECENCY` in a data message is the same slot as `APP_KEY_MMOL` in a preferences message. I think here the condition should be on what type of message it is. If it's a [data message](https://github.com/mddub/nightscout-graph-pebble/blob/77c70a/src/main.c#L96), you can assume recency is present, and do this calculation for the next update time. If it's an error message (phone couldn't reach NS), it should (I think) retry in the default 60 seconds.